lib/external_config/s3.ex

```diff
@@ -30,4 +30,40 @@ defmodule ExternalConfig.S3 do
         {nil, %{}}
     end
   end
+
+  def get_active_headend_ip() do
+    s3_client = Application.get_env(:realtime_signs, :s3_client)
+    aws_client = Application.get_env(:realtime_signs, :aws_client)
+    bucket = Application.get_env(:realtime_signs, :s3_bucket)
+    path = Application.get_env(:realtime_signs, :s3_active_headend_path)
+
+    case s3_client.get_object(bucket, path)
+         |> aws_client.request() do
+      {:ok, response} ->
+        body = Jason.decode!(response.body)
+        {:ok, body["active_headend_ip"]}
+
+      {:error, e} ->
+        Logger.error("active_headend_ip: s3 response error: #{inspect(e)}")
+        {:error, nil}
+    end
+  end
+
+  def put_active_headend_ip(ip) do
+    s3_client = Application.get_env(:realtime_signs, :s3_client)
+    aws_client = Application.get_env(:realtime_signs, :aws_client)
+    bucket = Application.get_env(:realtime_signs, :s3_bucket)
+    path = Application.get_env(:realtime_signs, :s3_active_headend_path)
+
+    case s3_client.put_object(bucket, path, Jason.encode!(%{active_headend_ip: ip}))
+         |> aws_client.request() do
+      {:ok, response} ->
+        Logger.info("active_headend_ip: config changed to: #{ip}")
+        {:ok, response}
```

Comment:
If we don't set the config value here, this won't take effect for 10 to 20 seconds until the next scheduled reads happen.

Reply:
Right, but we need the delay there if we want to have the "two consecutive" check in play. If we end up removing that, then we would update the config right away here.